cmd/fileshare_process/main.go

```diff
@@ -0,0 +1,108 @@
+// NordVPN fileshare daemon.
+package main
+
+import (
+	"errors"
+	"log"
+	_ "net/http/pprof" // #nosec G108 -- http server is not run in production builds
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+
+	"github.com/NordSecurity/nordvpn-linux/fileshare_process"
+	"github.com/NordSecurity/nordvpn-linux/fileshare_startup"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+)
+
+// Values set when building the application
+var Environment = ""
+
+func openLogFile(path string) (*os.File, error) {
+	// #nosec path is constant
+	logFile, err := os.OpenFile(path, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0600)
+	if err != nil {
+		return nil, err
+	}
+	return logFile, nil
+}
+
+func main() {
+	if logFile, err := openLogFile(filepath.Join(fileshare_process.FileshareLogPath, "/nordfileshared.log")); err == nil {
+		log.SetOutput(logFile)
+		log.SetFlags(log.LstdFlags | log.Lshortfile | log.Lmicroseconds)
+	}
+
+	processStatus := fileshare_process.GRPCFileshareProcess{}.ProcessStatus()
+	if processStatus == fileshare_process.Running {
+		os.Exit(int(fileshare_process.CodeAlreadyRunning))
+	} else if processStatus == fileshare_process.RunningForOtherUser {
+		os.Exit(int(fileshare_process.CodeAlreadyRunningForOtherUser))
+		log.Println("Cannot start fileshare daemon, it is already running for another user.")
```

Comment:
Is message getting logged?

Reply:
I think so, logger is initialized before.

Reply:
but there is an `os.Exit` above it. isn't that stopping immediately the process?